core/src/androidMain/kotlin/Scanner.kt

```diff
@@ -53,22 +56,34 @@ public class AndroidScanner internal constructor(
             }
 
             override fun onScanFailed(errorCode: Int) {
+                logger.error { message = "Scan could not be started, error code $errorCode." }
                 cancel("Bluetooth scan failed", ScanFailedException(errorCode))
             }
         }
 
-        val scanFilter =
-            filterServices
-                ?.map { ScanFilter.Builder().setServiceUuid(ParcelUuid(it)).build() }
-                ?.toList()
-        bluetoothAdapter.bluetoothLeScanner.startScan(
-            scanFilter,
-            scanSettings,
-            callback,
-        )
+        val scanFilter = filterServices
+            ?.map { ScanFilter.Builder().setServiceUuid(ParcelUuid(it)).build() }
+            ?.toList()
+        logger.info {
+            message = when (filterServices) {
+                null -> "Starting scan with no service filter."
+                else -> "Starting scan for services ${filterServices.joinToString()}."
+            }
+        }
+        scanner.startScan(scanFilter, scanSettings, callback)
 
         awaitClose {
-            bluetoothAdapter.bluetoothLeScanner.stopScan(callback)
+            logger.info {
+                message = when (filterServices) {
+                    null -> "Stopping scan with no service filter."
+                    else -> "Stopping scan for services ${filterServices.joinToString()}."
+                }
+            }
+            try {
+                scanner.stopScan(callback)
+            } catch (e: IllegalStateException) {
+                logger.warn { message = "Failed to stop scan." }
```

Comment:
Does Kable's logging allow plugging in the exception?

Reply:
I'm pretty sure it does.